python-package/xgboost/sklearn.py

```diff
@@ -749,7 +749,10 @@ def fit(self, X, y, sample_weight=None, base_margin=None,
         # pylint: disable = attribute-defined-outside-init,arguments-differ
 
         evals_result = {}
-        self.classes_ = np.unique(y)
+        if self.objective != 'binary:logistic':
+            self.classes_ = np.array([0.0, 1.0])
```

Comment:
??? Why 0 and 1 here?

Reply:
@hcho3  Sorry, still writing tests.  Just raised the PR early.